### PR TITLE
[New Rule] Adding Coverage for `AWS Discovery API Calls via CLI from a Single Resource`

### PR DESCRIPTION
**Pull request by terrancedejesus**
# Pull Request

*Issue link(s)*:
* https://github.com/elastic/ia-trade-team/issues/456
* https://github.com/elastic/ia-trade-team/issues/272

## Summary - What I changed
Added detection coverage for `AWS Discovery API Calls via CLI from a Single Resource`.

Detects when a single AWS resource is running multiple `Describe` and `List` API calls in a 30-second window. This
behavior could indicate an actor attempting to discover the AWS infrastructure using compromised credentials or a
compromised instance. Adversaries may use this information to identify potential targets for further exploitation or to
gain a better understanding of the target's infrastructure.

## How To Test
Screenshot 2024-11-04 at 4 23 26 PM

## Checklist

- [x] Added a label for the type of pr: `bug`, `enhancement`, `schema`, `Rule: New`, `Rule: Deprecation`, `Rule: Tuning`, `Hunt: New`, or `Hunt: Tuning` so guidelines can be generated
- [ ] Added the `meta:rapid-merge` label if planning to merge within 24 hours
- [x] Secret and sensitive material has been managed correctly
- [x] Automated testing was updated or added to match the most common scenarios
- [ ] Documentation and comments were added for features that require explanation

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

**Comment by protectionsmachine** [comment]
## Rule: New - Guidelines

These guidelines serve as a reminder set of considerations when proposing a new rule.

### Documentation and Context

- [ ] Detailed description of the rule.
- [ ] List any new fields required in ECS/data sources.
- [ ] Link related issues or PRs.
- [ ] Include references.

### Rule Metadata Checks

- [ ] `creation_date` matches the date of creation PR initially merged.
- [ ] `min_stack_version` should support the widest stack versions.
- [ ] `name` and `description` should be descriptive and not include typos.
- [ ] `query` should be inclusive, not overly exclusive, considering performance for diverse environments. Non ecs fields should be added to `non-ecs-schema.json` if not available in an integration.
- [ ] `min_stack_comments` and `min_stack_version` should be included if the rule is only compatible starting from a specific stack version.
- [ ] `index` pattern should be neither too specific nor too vague, ensuring it accurately matches the relevant data stream (e.g., use logs-endpoint.process-* for process data).
- [ ] `integration` should align with the `index`. If the integration is newly introduced, ensure the manifest, schemas, and `new_rule.yaml` template are updated.
- [ ] `setup` should include the necessary steps to configure the integration.
- [ ] `note` should include any additional information (e.g. Triage and analysis investigation guides, timeline templates).
- [ ] `tags` should be relevant to the threat and align/added to the `EXPECTED_RULE_TAGS` in the definitions.py file.
- [ ] `threat`, `techniques`, and `subtechniques` should map to ATT&CK always if possible.

#### New BBR Rules
- [ ] `building_block_type` should be included if the rule is a building block and the rule should be located in the `rules_building_block` folder.
- [ ] `bypass_bbr_timing` should be included if adding custom lookback timing to the rule.

### Testing and Validation

- [ ] Provide evidence of testing and detecting the expected threat.
- [ ] Check for existence of coverage to prevent duplication.

File: rules/integrations/aws/discovery_ec2_multiple_discovery_api_calls_via_cli.toml [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    "Tactic: Discovery",
+]
+timestamp_override = "event.ingested"
+type = "esql"
+
+query = '''
+from logs-aws.cloudtrail*
+
+// create time window buckets of 10 seconds
+| eval time_window = date_trunc(10 seconds, @timestamp)
```
Comment (imays11) [inline rules/integrations/aws/discovery_ec2_multiple_discovery_api_calls_via_cli.toml:85]: 30 seconds in the description but 10 seconds in the query

File: rules/integrations/aws/discovery_ec2_multiple_discovery_api_calls_via_cli.toml [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    event.dataset == "aws.cloudtrail"
+
+    // filter on CloudTrail audit logs for IAM, EC2, and S3 events only
+    and event.provider in ("iam.amazonaws.com", "ec2.amazonaws.com", "s3.amazonaws.com")
+
+    // ignore AWS service actions
+    and aws.cloudtrail.user_identity.type != "AWSService"
+
+    // filter for aws-cli specifically
+    and user_agent.name == "aws-cli"
```
Comment (imays11) [inline rules/integrations/aws/discovery_ec2_multiple_discovery_api_calls_via_cli.toml:96]: Why just the cli? what about `Boto3` user-agent for python scripts?

File: rules/integrations/aws/discovery_ec2_multiple_discovery_api_calls_via_cli.toml [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    // filter out AWS, Elastic, Datadog, and Wiz service roles by ARN
+    aws_vendor = starts_with(actor, "AWSServiceRoleForConfig"),
+    elastic_vendor = starts_with(actor, "Elastic"),
+    datadog_vendor = starts_with(actor, "DataDog"),
+    wiz_vendor = starts_with(actor, "Wiz")
+| where
+    aws_vendor != true and
+    elastic_vendor != true and
+    datadog_vendor != true and
+    wiz_vendor != true
```
Comment (imays11) [inline rules/integrations/aws/discovery_ec2_multiple_discovery_api_calls_via_cli.toml:121]: Should we leave these type of exclusions up to users?

File: rules/integrations/aws/discovery_ec2_multiple_discovery_api_calls_via_cli.toml [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+    and user_agent.name == "aws-cli"
+
+    // exclude DescribeCapacityReservations events related to AWS Config
+    and not event.action in ("DescribeCapacityReservations")
+
+    // filter for Describe and List API calls
+    and (
+            event.action like "Describe*" or
+            event.action like "List*" or
+            event.action like "Get*"
```
Comment (Aegrah) [inline rules/integrations/aws/discovery_ec2_multiple_discovery_api_calls_via_cli.toml:105]: Given that, we know what `event.actions` we are talking about here, I think we should rather add a large list than a wildcard.

**Review by Aegrah — APPROVED** [review]

File: rules/integrations/aws/discovery_ec2_multiple_discovery_api_calls_via_cli.toml [hunk 5]
```diff
@@ -0,0 +1,147 @@
+[metadata]
+creation_date = "2024/11/04"
+integration = ["aws"]
+maturity = "production"
+updated_date = "2024/11/04"
+
+[rule]
+author = ["Elastic"]
+description = """
+Detects when a single AWS resource is running multiple `Describe` and `List` API calls in a 30-second window. This
```
Comment (terrancedejesus) [inline rules/integrations/aws/discovery_ec2_multiple_discovery_api_calls_via_cli.toml:10]:
```suggestion
Detects when a single AWS resource is running multiple `Describe` and `List` API calls in a 10-second window. This
```

File: rules/integrations/aws/discovery_ec2_multiple_discovery_api_calls_via_cli.toml (hunk 2 above)
Comment (terrancedejesus) [reply rules/integrations/aws/discovery_ec2_multiple_discovery_api_calls_via_cli.toml:96]: SDK's require fetching credentials and adding them into a script or tool of sorts.
AWS CLI can be OOTB and prime for opportunistic TAs that just landed on an EC2 instance box.

Both are plausible, but the SDK route requires additional steps and alerts would likely be FP from admins or expected automation or scheduled jobs.

SCARLETEEL, Midnight Blizzard and Scattered Spider were known to use the AWS CLI installed on the compromised instance for discovery and enumeration.

File: rules/integrations/aws/discovery_ec2_multiple_discovery_api_calls_via_cli.toml (hunk 3 above)
Comment (terrancedejesus) [reply rules/integrations/aws/discovery_ec2_multiple_discovery_api_calls_via_cli.toml:121]: Agreed, as discussed I will remove them, however specifically ignore the AWSServiceRoleForConfig as we can be specific about the ARN path which cannot be duplicated but is known to generate FPs.

File: rules/integrations/aws/discovery_ec2_multiple_discovery_api_calls_via_cli.toml (hunk 4 above)
Comment (terrancedejesus) [reply rules/integrations/aws/discovery_ec2_multiple_discovery_api_calls_via_cli.toml]: Added 

```sql
| where true in (
    starts_with(event.action, "Describe"),
    starts_with(event.action, "Get"),
    starts_with(event.action, "List"),
    starts_with(event.action, "Generate")
)
```

Which should be more performant. Also expanded providers so the API call list would have expanded exponentially.

**Review by shashank-elastic — APPROVED** [review]

**Review by imays11 — APPROVED** [review]
Looks good!

**Pull request merged**